test/e2e/util.go

```diff
@@ -17,12 +18,15 @@ func ReadTestCatalogServerContents(ctx context.Context, catalog *catalogd.Cluste
 	if catalog == nil {
 		return nil, fmt.Errorf("cannot read nil catalog")
 	}
-	url, err := url.Parse(catalog.Status.ContentURL)
+	if catalog.Status.URLs == nil {
+		return nil, fmt.Errorf("catalog %s has no catalog urls", catalog.Name)
```

Comment:
Best practice in almost all situations is to use %q to get quoting of strings. That way if (somehow) `catalog.Name` is an empty string, we get `catalog "" has no catalog urls` instead of `catalog  has no catalog urls`.

```suggestion
		return nil, fmt.Errorf("catalog %q has no catalog urls", catalog.Name)
```